cagov/ui-claim-tracker
# Add appointment to Claim Status summary

## Ticket

Resolves #341 

## Changes

- Create `<Appointment>` component (with story and tests)
- Create `<ClaimSummary>` component (with story and tests)
- Add business logic for when to display an appointment as part of Scenario 2
- Add translation strings for appointments
- Add missing Scenario 2 and Scenario 3 snapshot tests for `<ClaimStatus>` component
- Add test helpers (broken out of `getScenarioContent.test.tsx`)
- Fix bugs in formatDate
- Fix bugs in timeSlot
- Add a util file for handling Sentence case in strings

## Context

This PR is big, but it is only actually handling one very narrow case: when to display the appointment date (`pendingDetermination.scheduleDate`) and time (`pendingDetermination.timeSlotDesc`) and how to format it.

## Testing

`yarn test`

ALSO

1. While #299 is not resolved, comment out the following lines: https://github.com/cagov/ui-claim-tracker/blob/32990f892c6b84017c30af5d273fdd94410649e7/pages/index.tsx#L95-L98
2. Run `yarn storybook`
3. Navigate to Page > Default 
4. Select scenario "Determination interview: scheduled"
5. **Verify it looks right**
6. Click the globe in the top menu bar and select "Español"
7. **Verify it looks right**
8. Navigate to Atoms > Appointment
9. Select a date (ignore the time picker) and **verify** in both English and Spanish
10. Enter a start and end time (must be between 1–12) and **verify** in both English and Spanish

Note: that some Spanish translation strings are still missing from `main` at the moment, but will be added with #329.
